optimizely/event/event_processor.py

```diff
@@ -54,12 +56,14 @@ def __init__(self,
                 logger,
                 default_start=False,
                 event_queue=None,
+                notification_center=None,
                 batch_size=None,
                 flush_interval=None,
                 timeout_interval=None):
     self.event_dispatcher = event_dispatcher
     self.logger = logger
     self.event_queue = event_queue or queue.Queue(maxsize=self._DEFAULT_QUEUE_CAPACITY)
+    self.notification_center = notification_center
```

Comment:
what if notification center is not set.

Reply:
Default value for notification_center is set to None. Before sending out a notification, the condition for it being not None is checked. 

`notification_center=None,`

```
if self.notification_center is not None:
      self.notification_center.send_notifications(
        enums.NotificationTypes.LOG_EVENT,
        log_event
      )
```